prov/efa/src/rxr/rxr_init.c

```diff
@@ -552,6 +552,11 @@ static void rxr_fini(void)
 		if (shm_info)
 			fi_freeinfo(shm_info);
 	}
+
+#if HAVE_EFA_DL
+	ofi_monitor_cleanup();
+	ofi_mem_fini();
+#endif
```

Comment:
We've had this same problem in other cases.  Maybe there's some generic way to handle this that all providers could use?  (Don't let that suggestion prevent merging this.)

Reply:
Yeah, I couldn't find a best-practice kind of way to handle this with some quick research I did on linker behaviors. This change in itself is assuming that the init functions are idempotent (but they are not really, the monitor init defines parameters).

I'll take an action item to find a better way to handle this, but will merge this for now to plug outstanding issue.